core/actions/data_preparation.ts

```diff
@@ -128,6 +131,64 @@ function parseDataPreparationDefinitionJson(dataPreparationAsJson: {
   }
 }
 
+function resolveSourcesAndDestinations(
+    actionBuilder: ActionBuilder<dataform.DataPreparation>,
+    definition: dataform.dataprep.DataPreparation
+): dataform.dataprep.DataPreparation {
+  const resolvedDataPreparation = dataform.dataprep.DataPreparation.create(definition);
+
+  // Resolve error table, if set
+  const errorTable = definition.configuration?.errorTable;
+  if (errorTable) {
+    resolvedDataPreparation.configuration.errorTable =
+        applySessionDefaultsToTableReference(actionBuilder, errorTable);
+  }
+
+  // Loop through all nodes and resolve the compilation overrides for
+  // all source and destination tables.
+  definition.nodes.forEach((node, index) => {
+
+        // Resolve source tables, if set.
+        const sourceTable = node.source.table;
+        if (sourceTable) {
+          resolvedDataPreparation.nodes[index].source.table =
+              applySessionDefaultsToTableReference(actionBuilder, sourceTable);
+        }
+
+        // Resolve destination tables, if set.
+        const destinationTable = node.destination?.table;
+        if (destinationTable) {
+          resolvedDataPreparation.nodes[index].destination.table =
+              applySessionDefaultsToTableReference(actionBuilder, destinationTable);
+        }
+      }
+
+    );
+
+  return resolvedDataPreparation;
+}
+
+function applySessionDefaultsToTableReference(
```

Comment:
To mirror the naming of `applySessionToTarget`, this would better be `applySessionToTableReference` instead

Reply:
Done.